lib/importers/category_importer.rb

```diff
@@ -18,42 +19,49 @@ def initialize(wiki, opts={})
     @max_wp10 = opts[:max_wp10] || 100
   end
 
+  def mainspace_page_titles_for_category(category, depth=0)
+    CategoryUtils.get_titles_without_prefixes(page_data_for_category(category, depth))
+  end
+
   def page_titles_for_category(category, depth=0, namespace=nil)
-    page_properties_for_category(category, 'title', depth, namespace)
+    page_data_for_category(category, depth, namespace, 'title')
   end
 
   private
 
-  def page_properties_for_category(category, property, depth=0, namespace=nil)
+  def page_data_for_category(category, depth=0, namespace=nil, property=nil)
     cat_query = category_query(category, namespace)
-    page_data = get_category_member_properties(cat_query, property)
+    page_data = get_category_member_data(cat_query, property)
     if depth.positive?
       depth -= 1
       subcats = subcategories_of(category)
       subcats.each do |subcat|
-        page_data += page_properties_for_category(subcat, property, depth, namespace)
+        page_data += page_data_for_category(subcat, depth, namespace, property)
       end
     end
     page_data
   end
 
-  def get_category_member_properties(query, property)
-    property_values = []
+  def get_category_member_data(query, property)
+    data_values = []
     continue = true
     until continue.nil?
       cat_response = WikiApi.new(@wiki).query query
       page_data = cat_response.data['categorymembers']
-      page_data.each { |page| property_values << page[property] }
+      page_data.each do |page|
+        data = property.present? ? page[property] : page
```

Comment:
This is a bit confusing. In what situation will the requested property no be present?

Reply:
Earlier property data was getting filtered here itself, and with that I had to combine an implementation which required the full page data (to know the namespace and accordingly remove the prefix to fix the bug), but all of this was getting unnecessarily confusing. Now all the data received gets returned and whatever property filtering required is done in `CategoryUtils`, as filtering that data shouldn't really be the job of `get_category_member_data`.